addons/mail/models/discuss/ir_binary.py

```diff
@@ -10,15 +11,16 @@ def _find_record_check_access(self, record, access_token, field):
         if record._name in ["res.partner", "mail.guest"] and field == "avatar_128":
             current_partner, current_guest = self.env["res.partner"]._get_current_persona()
             if current_partner or current_guest:
-                domain = []
+                member_lookup_domains = []
                 if record._name == "res.partner":
-                    domain.append(("channel_member_ids", "any", [("partner_id", "=", record.id)]))
+                    member_lookup_domains.append([("partner_id", "=", record.id)])
                 else:
-                    domain.append(("channel_member_ids", "any", [("guest_id", "=", record.id)]))
+                    member_lookup_domains.append([("guest_id", "=", record.id)])
                 if current_partner:
-                    domain.append(("channel_member_ids", "any", [("partner_id", "=", current_partner.id)]))
+                    member_lookup_domains.append([("partner_id", "=", current_partner.id)])
                 else:
-                    domain.append(("channel_member_ids", "any", [("guest_id", "=", current_guest.id)]))
+                    member_lookup_domains.append([("guest_id", "=", current_guest.id)])
+                domain = [('channel_member_ids', 'any', OR(member_lookup_domains))]
```

Comment:
```suggestion
                domain = [('channel_member_ids', 'any', AND(member_lookup_domains))]
```

Reply:
Fix has been pushed, and benchmarks have been updated. (same timing when the domain is not falsy, but now we got a good improvement if the domain is falsy, it's instant 🚀 ).
I f*ed up the original domain was a conjunction, not a disjonction, my bad. Thanks for pointing this out :)